File: Affine/affine.py
LETTER = 'abcdefghijklmnopqrstuvwxyz'
# key(a,b)
N = 26 # Module

def gcd(number_1, number_2):
	if number_1 < number_2:
		tmp = number_1
		number_1 = number_2
		number_2 = tmp

	if number_1 % number_2 == 0:
		return number_2
	else:
		return gcd(number_2, number_1%number_2)

def inverseModule(number, module, x=0):
	if (module*x+1) % number == 0:
		return int((module*x+1)/number)
	else:
		return int(inverseModule(number, module, x+1))

def affineEncrypt(plaintext, a, b):
	text = ''
	
	for t in plaintext:
		text += LETTER[(a * LETTER.index(t) + b) % N]
	
	return text

def affineDecrypt(ciphertext, a, b):
	text = ''
	inverse_a = inverseModule(a, N);

	for t in ciphertext:
		text += LETTER[(inverse_a * (LETTER.index(t) - b) ) % N]
	
	return text

def main():

	while True:
		print("Key(a,b)=>")
		a = int(input("a: "))
		b = int(input("b: "))
	
		if gcd(a, N) == 1:
			text = input("Text: ")
			selection = int(input("Enter 0 to Encrypt - 1 to Decrypt:"))
			if selection == 0:
				print(affineEncrypt(text, a, b))
				break
			else:
				print(affineDecrypt(text, a, b))
				break
		else:
			print("Key invalid!")

main()

File: Affine/affine.cpp
#include <iostream>
#include <string>
using namespace std;

string Encrypt(int a, int b, string x) {
   string y;
   for (int i = 0; i < x.length(); i++)
      y += (char)((a * (x[i]-'a') + b+26) % 26+'a');
   return y;
}

string Decrypt(int inverse_a, int b, string y) {
   string x;
   for (int i = 0; i < y.length(); i++) {
      x = x + (char)(inverse_a * ((int)(y[i] - 'a') - b + 26) % 26 + 'a');
   }
   return x;
}
int modInverse(int a, int m) {
   for (int x = 1; x < m; x++)
      if (((a % m) * (x % m)) % m == 1) return x;
   return 1;
}

int main() {
   int a, b, c;
   // a = [1,3,5,7,9,11,15,17,19,21,23,25]
   string str;
   cout << "\nInsert Key: a=";
   cin >> a;
   cout << "\nInsert Key: b=";
   cin >> b;
   cout << "\nEnter string: ";
   cin >> str;
   cout << "\nType 0 to Encrypt, type 1 to Decrypt: ";

   cin >> c;
   if (c)
      cout << Decrypt(modInverse(a, 26), b, str);
   else
      cout << Encrypt(a, b, str);

   return 0;
}